allium/lib/bandwidth_formatter.py

```
@@ -144,36 +144,40 @@ def format_bandwidth_with_suffix(self, bandwidth_bytes, unit=None, decimal_place
         return f"{value} {unit}"
 
 
+# Cached formatter instances (BandwidthFormatter has no mutable state, safe to reuse)
+_formatter_bits = BandwidthFormatter(use_bits=True)
+_formatter_bytes = BandwidthFormatter(use_bits=False)
+
+
+def _get_formatter(use_bits=False):
```

Ruff (0.15.2)

[warning] Missing return type annotation for private function `_get_formatter`

(ANN202)